backport_packages/setup_backport_packages.py

```diff
@@ -422,40 +449,680 @@ def usage():
+                                    expected_ancestor=Protocol)
+    secrets = find_all_subclasses(expected_package=full_package_name,
+                                  expected_ancestor=BaseSecretsBackend)
+    new_operators, moved_operators = get_new_and_moved_classes(operators, MOVED_OPERATORS_DICT)
+    new_sensors, moved_sensors = get_new_and_moved_classes(sensors, MOVED_SENSORS_DICT)
+    new_hooks, moved_hooks = get_new_and_moved_classes(hooks, MOVED_HOOKS_DICT)
+    new_protocols, moved_protocols = get_new_and_moved_classes(protocols, MOVED_PROTOCOLS_DICT)
+    new_secrets, moved_secrets = get_new_and_moved_classes(secrets, MOVED_SECRETS_DICT)
+    class_summary = {
+        "NEW_OPERATORS": new_operators,
+        "MOVED_OPERATORS": moved_operators,
+        "NEW_SENSORS": new_sensors,
+        "MOVED_SENSORS": moved_sensors,
+        "NEW_HOOKS": new_hooks,
+        "MOVED_HOOKS": moved_hooks,
+        "NEW_PROTOCOLS": new_protocols,
+        "MOVED_PROTOCOLS": moved_protocols,
+        "NEW_SECRETS": new_secrets,
+        "MOVED_SECRETS": moved_secrets,
+    }
+    for from_name, to_name, object_type in [
+        ("NEW_OPERATORS", "NEW_OPERATORS_TABLE", "operators"),
+        ("NEW_SENSORS", "NEW_SENSORS_TABLE", "sensors"),
+        ("NEW_HOOKS", "NEW_HOOKS_TABLE", "hooks"),
+        ("NEW_PROTOCOLS", "NEW_PROTOCOLS_TABLE", "protocols"),
+        ("NEW_SECRETS", "NEW_SECRETS_TABLE", "secrets"),
+    ]:
+        class_summary[to_name] = convert_new_classes_to_table(class_summary[from_name],
+                                                              full_package_name,
+                                                              object_type)
+    for from_name, to_name, object_type in [
+        ("MOVED_OPERATORS", "MOVED_OPERATORS_TABLE", "operators"),
+        ("MOVED_SENSORS", "MOVED_SENSORS_TABLE", "sensors"),
+        ("MOVED_HOOKS", "MOVED_HOOKS_TABLE", "hooks"),
+        ("MOVED_PROTOCOLS", "MOVED_PROTOCOLS_TABLE", "protocols"),
+        ("MOVED_SECRETS", "MOVED_SECRETS_TABLE", "protocols"),
+    ]:
+        class_summary[to_name] = convert_moved_objects_to_table(class_summary[from_name],
+                                                                full_package_name,
+                                                                object_type)
+    return class_summary
+
+
+def render_template(template_name: str, context: Dict[str, Any]) -> str:
+    """
+    Renders template based on it's name. Reads the template from <name>_TEMPLATE.md.jinja2 in current dir.
+    :param template_name: name of the template to use
+    :param context: Jinja2 context
+    :return: rendered template
+    """
+    from jinja2 import Template
+    template_file_path = os.path.join(MY_DIR_PATH, f"{template_name}_TEMPLATE.md.jinja2")
+    with open(template_file_path, "rt") as template_file:
+        # remove comment
+        template = Template(template_file.read(), autoescape=True)
```

Comment:
By default, Jinja is a dangerous default undefined value. When an undefined variable is used, the empty value is displayed. 
Here is an example of how to render templates more securely.
```
def render_template(template_name: str, *args, **kwargs) -> str:
    """Render Jinja template"""
    TEMPLATE_LOADER = jinja2.FileSystemLoader(searchpath=TPL_PATH)
	TEMPLATE_ENV = jinja2.Environment(
	    loader=TEMPLATE_LOADER, undefined=jinja2.StrictUndefined, trim_blocks=True, lstrip_blocks=True
	)
    template = TEMPLATE_ENV.get_template(template_name)
    content: str = template.render(*args, **kwargs)
    return content
```
It's may be related to: https://github.com/apache/airflow/pull/8807#discussion_r425580282 but I'm not sure.

Reply:
Good point. Yes it was bad rename of parameter !  Changed it to be strict undefined :)